src/spacel/provision/db/rds.py

```diff
@@ -0,0 +1,219 @@
+                                                    or 'Private')
+
+            rds_desc = '%s for %s in %s' % (name, app_name, orbit_name)
+            logger.debug('Creating database "%s".', name)
+
+            # Create a parameter for the database password:
+            password_param = '%sPassword' % rds_resource
+            params[password_param] = {
+                'Type': 'String',
+                'Description': 'Password for database %s' % name,
+                'NoEcho': True
+            }
+
+            # Security group for database:
+            rds_sg_resource = '%sSg' % rds_resource
+            resources[rds_sg_resource] = {
+                'Type': 'AWS::EC2::SecurityGroup',
+                'Properties': {
+                    'GroupDescription': rds_desc,
+                    'VpcId': {'Ref': 'VpcId'},
+                    'SecurityGroupIngress': [
+                        {
+                            'IpProtocol': 'tcp',
+                            'FromPort': db_port,
+                            'ToPort': db_port,
+                            'SourceSecurityGroupId': {'Ref': 'Sg'}
+                        }
+                    ]
+                }
+            }
+
+            rds_params = {
+                'AllocatedStorage': storage_size,
+                'AllowMajorVersionUpgrade': False,
+                'AutoMinorVersionUpgrade': True,
+                'DBInstanceClass': instance_type,
+                'DBName': name,
+                'DBSubnetGroupName': {'Ref': db_subnet_group},
+                'Engine': db_type,
+                'EngineVersion': db_version,
+                'MasterUsername': db_username,
+                'MasterUserPassword': {'Ref': password_param},
+                'MultiAZ': multi_az,
+                'Port': db_port,
+                'PubliclyAccessible': public,
+                'StorageEncrypted': encrypted,
+                'StorageType': storage_type,
+                'VPCSecurityGroups': [{'Ref': rds_sg_resource}]
+            }
+
+            if storage_iops:
+                rds_params['Iops'] = storage_iops
+                if storage_type != 'io1':
+                    logger.warn('Overriding "storage_type" of "%s": ' +
+                                '"iops" requires io1.', name)
+                    rds_params['StorageType'] = 'io1'
+
+            # Workaround for the instance_type default not supporting crypt
+            # Other t2's fail, but at least that's the user's fault.
+            if encrypted and instance_type == 'db.t2.micro':
+                logger.warn('Overriding "instance_type" of "%s": ' +
+                            '"encrypted" requires t2.large".', name)
+                rds_params['DBInstanceClass'] = 'db.t2.large'
+
+            resources[rds_resource] = {
+                'Type': 'AWS::RDS::DBInstance',
+                'Properties': rds_params
+            }
+
+            encrypted, plaintext_func = \
```

Comment:
Returning the "promise" for the plaintext password is the grossest bit of this to me.